### PR TITLE
Names for some fields and methods in net.minecraft.data.server

### PR DESCRIPTION
**Pull request by bluebear94**
Some of the names might be up for debate so feel free to criticize.

### CODE REVIEW

File: mappings/net/minecraft/data/server/BlockLootTableGenerator.mapping [hunk 1]
```diff
@@ -1 +1,67 @@
 CLASS net/minecraft/class_2430 net/minecraft/data/server/BlockLootTableGenerator
+	FIELD field_11336 NEED_SILK_TOUCH Lnet/minecraft/class_4570$class_210;
+	FIELD field_11337 DOESNT_NEED_SILK_TOUCH Lnet/minecraft/class_4570$class_210;
+	FIELD field_11338 JUNGLE_SAPLING_DROP_CHANCES_FROM_LEAVES [F
+	FIELD field_11339 SAPLING_DROP_CHANCES_FROM_LEAVES [F
+	FIELD field_11340 ALWAYS_DROPPED_FROM_EXPLOSION Ljava/util/Set;
+	FIELD field_11341 DOESNT_NEED_SILK_TOUCH_SHEARS Lnet/minecraft/class_4570$class_210;
+	FIELD field_11342 NEED_SILK_TOUCH_SHEARS Lnet/minecraft/class_4570$class_210;
+	FIELD field_11343 NEED_SHEARS Lnet/minecraft/class_4570$class_210;
```
Comment (Juuxel) [inline mappings/net/minecraft/data/server/BlockLootTableGenerator.mapping]: I think `NEEDS_*` would be better for the `NEED_*` conditions.

**Review by Juuxel — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/data/server/BlockLootTableGenerator.mapping (hunk 1 above)
Comment (bluebear94) [reply mappings/net/minecraft/data/server/BlockLootTableGenerator.mapping]: Changed to `NEEDS_*`. I also changed the `REQUIRES_*` constants in `FishingLootTableGenerator` to `NEEDS_*` for consistency.

**Review by modmuss50 — APPROVED** [review]

**Review by liach — COMMENTED** [review]
Imo you should simplify names, e.g. in "block loot table" class your method/field names need not include "loot table" or "block", such as `createLootTableForBlockNeedingShears` -> `createForNeedingShears`, etc.

**Comment by bluebear94** [comment]
> Imo you should simplify names, e.g. in "block loot table" class your method/field names need not include "loot table" or "block", such as `createLootTableForBlockNeedingShears` -> `createForNeedingShears`, etc.

Are the names in 4a8ab4a better?

**Comment by bluebear94** [comment]
Also fixed the names in `EntityLootTableGenerator` (and fixed a rogue `NEED_*` there).

**Review by liach — APPROVED** [review]
Names are concise enough IMO.

**Pull request merged**